src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const DAMAGE = 50;
+
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = DEFAULT_HEALTH) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= DAMAGE;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter(
+          (aliveAnimal) => aliveAnimal !== target,
+        );
+      }
+    }
```

Comment:
The bite method currently checks if the target is an instance of Herbivore and not hidden, which is correct. However, it should also check if the target is not an instance of Carnivore, according to the task requirements.